test/domainRegicter.js

```diff
@@ -0,0 +1,119 @@
+        await domainRegister.connect(other_address).registerDomain("example2.com", other_address.address, { value: registrationFee });
+        await domainRegister.connect(other_address).registerDomain("example3.com", other_address.address, { value: registrationFee });
+        await domainRegister.connect(other_address).registerDomain("example4.com", other_address.address, { value: registrationFee });
+
+        const filter = domainRegister.filters.DomainRegistered();
+        const events = await domainRegister.queryFilter(filter);
+
+        const sortedEvents = events.sort((a, b) => {
+               return a.args.registrationTime < b.args.registrationTime ? -1 : a.args.registrationTime > b.args.registrationTime ? 1 : 0;
+             });
+        const domainsWithRegistrationTimes = sortedEvents.map(event => {
+            const date = new Date(parseInt(event.args.registrationTime) * 1000).toISOString();
+            return `${event.args.domain} - ${date}`;
+        });
+        console.log("List of domains with registration dates:", domainsWithRegistrationTimes);
+    });
+
+    it("should allow a user to register a domain", async function () {
+        await domainRegister.connect(other_address).registerDomain("example.com", other_address.address, { value: registrationFee });
+
+        const domainInfo = await domainRegister.domains("example.com");
+        await expect(domainInfo.controller).to.equal(other_address.address);
+        await expect(domainInfo.registrationTime).to.be.gt(0);
+    });
+
+    it("returns a list of domains for a specific controller, sorted by registration time", async function () {
+        await domainRegister.connect(owner).registerDomain("example.com", owner.address, { value: registrationFee });
+        await domainRegister.connect(another_address).registerDomain("example1.com", another_address.address, { value: registrationFee });
+        await domainRegister.connect(other_address).registerDomain("example2.com", other_address.address, { value: registrationFee });
+        await domainRegister.connect(another_address).registerDomain("example3.com", another_address.address, { value: registrationFee });
+        await domainRegister.connect(other_address).registerDomain("example4.com", other_address.address, { value: registrationFee });
+
+        const controllerAddress = another_address.address;
+        const filter = domainRegister.filters.DomainRegistered(null, controllerAddress, null);
+        const events = await domainRegister.queryFilter(filter);
+        const sortedEvents = events.sort((a, b) => {
+            return a.args.registrationTime < b.args.registrationTime ? -1 : a.args.registrationTime > b.args.registrationTime ? 1 : 0;
+        });
+        const domainNames = sortedEvents.map(event => event.args.domain);
+        console.log(domainNames);
+    });
+
+    it("should emit an event upon registration", async function () {
```

Comment:
line 73 
declared emit event, but not tested it
another event should be tested also